src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -321,107 +300,107 @@ private bool TryLookupLengthOrCount(SyntaxNode node, TypeSymbol inputType, out P
-                    diagnostics.AddRange(bindingDiagnostics);
+                    var analyzedArguments = AnalyzedArguments.GetInstance();
+                    analyzedArguments.Arguments.Add(new BoundIndexOrRangeIndexerPatternValuePlaceholder(syntax, argType));
+                    var receiver = new BoundImplicitReceiver(syntax, receiverType);
+                    BoundExpression boundAccess = BindIndexerOrIndexedPropertyAccess(syntax, receiver, indexerGroup, analyzedArguments, bindingDiagnostics);
+                    switch (boundAccess)
+                    {
+                        case BoundIndexerAccess { ResultKind: LookupResultKind.Viable, Indexer: { GetMethod: { } getMethod } } boundIndexerAccess when IsAccessible(getMethod, ref useSiteInfo):
+                            lookupResult.Clear();
+                            if (!TryLookupLengthOrCount(receiverType, lookupResult, out lengthProperty, ref useSiteInfo))
+                                goto default;
+                            ReportDiagnosticsIfObsolete(diagnostics, lengthProperty, syntax, hasBaseReceiver: false);
+                            // PROTOTYPE(list-patterns) Can this be ever true? If so, move to if above
+                            Debug.Assert(!boundIndexerAccess.Indexer.IsStatic);
+                            indexerAccess = BindIndexerDefaultArguments(boundIndexerAccess, BindValueKind.RValue, bindingDiagnostics);
+                            _ = GetWellKnownTypeMember(argIsIndex ? WellKnownMember.System_Index__ctor : WellKnownMember.System_Range__ctor, bindingDiagnostics, syntax: syntax);
+                            found = true;
+                            break;
+                        case BoundIndexOrRangePatternIndexerAccess boundIndexOrRangePatternIndexerAccess:
+                            patternSymbol = boundIndexOrRangePatternIndexerAccess.PatternSymbol;
+                            lengthProperty = boundIndexOrRangePatternIndexerAccess.LengthOrCountProperty;
+                            found = true;
+                            break;
+                        default:
+                            lengthProperty = null;
+                            found = false;
+                            break;
+                    }
+                    analyzedArguments.Free();
+                    indexerGroup.Free();
+                    goto done;
                 }
+                lookupResult.Clear();
+            }
 
-                analyzedArguments.Free();
-                indexerGroup.Free();
+            // If the argType is missing or the indexer lookup has failed, we will fallback to the implicit indexer support.
+            found = TryLookupLengthOrCount(receiverType, lookupResult, out lengthProperty, ref useSiteInfo) &&
+                    TryFindIndexOrRangeIndexerPattern(lookupResult, receiverOpt: null, receiverType, argIsIndex, out patternSymbol, diagnostics, ref useSiteInfo);
```

Comment:
I don't see any calls to `CheckImplicitThisCopyInReadOnlyMember` in `Binder_Patterns.cs` (but we have them when binding indexer access with Index or Range syntax). Are we missing them?

Reply:
We always capture the input in a temp so we could never mutate `this`.